### PR TITLE
Cleanup: Assimilate Scalarvalue into Prismavalue

### PR DESCRIPTION
**Pull request by do4gr**

### CODE REVIEW

File: migration-engine/migration-engine-tests/src/assertions.rs [hunk 1]
```diff
@@ -241,16 +241,11 @@ impl<'a> TableAssertion<'a> {
...
 impl<'a> ColumnAssertion<'a> {
-    pub fn assert_default(self, expected: Option<&str>) -> AssertionResult<Self> {
-        let found = self.0.default.as_ref().map(|default_value| match default_value {
-            DefaultValue::VALUE(s) => s,
-            DefaultValue::SEQUENCE(s) => s,
-            DefaultValue::DBGENERATED(s) => s,
-            DefaultValue::NOW => "CURRENT_TIMESTAMP",
-        });
+    pub fn assert_default(self, expected: Option<DefaultValue>) -> AssertionResult<Self> {
+        let found = &self.0.default;
```
Comment (tomhoule) [inline migration-engine/migration-engine-tests/src/assertions.rs:245]: :100:

File: libs/sql-schema-describer/src/postgres.rs [hunk 2]
```diff
@@ -779,12 +778,4 @@ mod tests {
...
         ));
     }
-    #[test]
-    fn postgres_unquote_string_default_regex_works() {
-        let quoted_str = "'abc $$ def'";
-
-        assert_eq!(unquote(quoted_str), "abc $$ def");
-
-        assert_eq!(unquote("heh "), "heh ");
-    }
```
Comment (tomhoule) [inline libs/sql-schema-describer/src/postgres.rs:789]: I think this test can be un-deleted if we use the function again

File: libs/datamodel/connectors/datamodel-connector/src/scalars.rs [hunk 3]
```diff
@@ -31,52 +28,9 @@ impl ToString for ScalarType {
...
-            ScalarValue::Int(val) => val.to_string(),
-            ScalarValue::Float(val) => val.to_string(),
-            ScalarValue::Decimal(val) => val.to_string(),
-            ScalarValue::Boolean(val) => val.to_string(),
-            ScalarValue::String(val) => val.to_string(),
-            ScalarValue::DateTime(val) => val.to_string(),
-            ScalarValue::ConstantLiteral(value) => value.to_string(),
-        }
-    }
-}
```
Comment (tomhoule) [inline libs/datamodel/connectors/datamodel-connector/src/scalars.rs:82]: :fire:

File: libs/datamodel/core/src/validator/lower.rs [hunk 4]
```diff
@@ -119,28 +120,29 @@ impl LowerDmlToAst {
...
-            dml::ScalarValue::Decimal(value) => ast::Expression::NumericValue(value.to_string(), ast::Span::empty()),
-            dml::ScalarValue::Float(value) => ast::Expression::NumericValue(value.to_string(), ast::Span::empty()),
-            dml::ScalarValue::Int(value) => ast::Expression::NumericValue(value.to_string(), ast::Span::empty()),
+    pub fn lower_prisma_value(pv: &PrismaValue) -> ast::Expression {
+        match pv {
+            PrismaValue::Boolean(true) => ast::Expression::BooleanValue(String::from("true"), ast::Span::empty()),
+            PrismaValue::Boolean(false) => ast::Expression::BooleanValue(String::from("false"), ast::Span::empty()),
+            PrismaValue::String(value) => ast::Expression::StringValue(value.clone(), ast::Span::empty()),
+            PrismaValue::Enum(value) => ast::Expression::ConstantValue(value.clone(), ast::Span::empty()),
+            PrismaValue::DateTime(value) => ast::Expression::ConstantValue(value.to_rfc3339(), ast::Span::empty()),
```
Comment (tomhoule) [inline libs/datamodel/core/src/validator/lower.rs:137]: I'm wondering if datetimes should be quoted or not. I don't think I've seen datetimes as constants in a schema before.

Comment (do4gr) [reply libs/datamodel/core/src/validator/lower.rs]: I adjusted it so they are lowered to string

**Pull request merged**